src/SpriteKit/SKWarpGeometryGrid.cs

```diff
@@ -29,30 +29,13 @@ public static SKWarpGeometryGrid Create (nint cols, nint rows, Vector2 [] source
 			if (destPositions.Length < ((cols + 1) * (rows + 1)))
 				throw new InvalidOperationException ("destPositions should have a minimum lenght of (cols + 1) * (rows + 1)");
 
-			IntPtr spPtr = IntPtr.Zero;
-			GCHandle spGch = new GCHandle ();
-			if (sourcePositions != null) {
-				spGch = GCHandle.Alloc (sourcePositions, GCHandleType.Pinned);
-				spPtr = spGch.AddrOfPinnedObject ();
-			}
-
-			IntPtr dpPtr = IntPtr.Zero;
-			GCHandle dpGch = new GCHandle ();
-			if (destPositions != null) {
-				dpGch = GCHandle.Alloc (destPositions, GCHandleType.Pinned);
-				dpPtr = dpGch.AddrOfPinnedObject ();
-			}
-
-			var grid = GridWithColumns (cols, rows, spPtr, dpPtr);
-
-			spGch.Free ();
-			dpGch.Free ();
-
-			return grid;
+			fixed (Vector2* source_ptr = &sourcePositions [0])
```

Comment:
Both sourcePositions and destPositions are guaranteed to have length > 0 due to argument validation earlier in the function.